### PR TITLE
ESQL: Plumb a way to run phased plans

### PR DESCRIPTION
**Pull request by nik9000**
INLINESTATS is going to run two ESQL commands - one to get the STATS and one to join the stats results to the output. This plumbs a way for `EsqlSession#execute` to run multiple dips into the compute engine using a `BiConsumer<PhysicalPlan, ActionListener<Result>> runPhase`. For now, we just plug that right into the output to keep things working as they are now. But soon, so soon, we'll plug in a second phase.

### CODE REVIEW

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks Nik!

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Result.java [hunk 1]
```diff
@@ -7,8 +7,10 @@
...
 package org.elasticsearch.xpack.esql.session;
 
+import org.elasticsearch.compute.data.Page;
+import org.elasticsearch.compute.operator.DriverProfile;
 import org.elasticsearch.xpack.esql.core.expression.Attribute;
 
 import java.util.List;
 
-public record Result(List<Attribute> columns, List<List<Object>> values) {}
+public record Result(List<Attribute> layout, List<Page> pages, List<DriverProfile> profiles) {}
```
Comment (alex-spies) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Result.java:16]: super nit: there's also a datatype `Layout` which is not quite just a list of attributes; we sometimes also use `fields` (`schema` would also be appropriate but isn't really used in our code base)

**Review by alex-spies — APPROVED** [review]
Neat!

**Review by ChrisHegarty — APPROVED** [review]
Noice! LGTM

**Comment by nik9000** [comment]
Thanks friends!

**Pull request merged**